src/main/java/stream/flarebot/flarebot/util/general/FormatUtils.java

```diff
@@ -0,0 +1,172 @@
+     *
+     * @param period Period to format onto the current date
+     * @return The date in a precise format. Example: 2017-10-13 21:56:33.681
+     */
+    public static String formatPrecisely(Period period) {
+        return FormatUtils.preciseFormat.format(DateTime.now(DateTimeZone.UTC).plus(period).toDate());
+    }
+
+    /**
+     * Formats a {@link LocalDateTime}
+     *
+     * @param dateTime The {@link LocalDateTime} to format.
+     * @return A String repressing the time.
+     */
+    public static String formatTime(LocalDateTime dateTime) {
+        LocalDateTime time = LocalDateTime.from(dateTime.atOffset(ZoneOffset.UTC));
+        return time.getDayOfMonth() + GeneralUtils.getDayOfMonthSuffix(time.getDayOfMonth()) + " " + time
+                .format(FormatUtils.timeFormat) + " UTC";
+    }
+
+    /**
+     * Formats time.
+     *
+     * @param duration The duration to format.
+     * @param durUnit The unit the duration is in.
+     * @param fullUnits Weather not not to return full units. ex days instead of d.
+     * @param append0 Weather or not to append 0s
+     * @return The formatted time.
+     */
+    public static String formatTime(long duration, TimeUnit durUnit, boolean fullUnits, boolean append0) {
+        long totalSeconds = 0;
+        switch (durUnit) {
+            case MILLISECONDS:
+                totalSeconds = duration / 1000;
+                break;
+            case SECONDS:
+                totalSeconds = duration;
+                break;
+            case MINUTES:
+                totalSeconds = duration * 60;
+                break;
+            case HOURS:
+                totalSeconds = (duration * 60) * 60;
+                break;
+            case DAYS:
+                totalSeconds = ((duration * 60) * 60) * 24;
+                break;
+        }
+        long seconds = totalSeconds % 60;
+        long minutes = (totalSeconds / 60) % 60;
+        long hours = (totalSeconds / 3600) % 24;
+        long days = (totalSeconds / 86400);
+        return (days > 0 ? (append0 && days < 10 ? "0" + days : days) + (fullUnits ? " days " : "d ") : "")
+                + (hours > 0 ? (append0 && hours < 10 ? "0" + hours : hours) + (fullUnits ? " hours " : "h ") : "")
+                + (minutes > 0 ? (append0 && minutes < 10 ? "0" + minutes : minutes) + (fullUnits ? " minutes" : "m ") : "")
+                + (seconds > 0 ? (append0 && seconds < 10 ? "0" + seconds : seconds) + (fullUnits ? " seconds" : "s") : "")
+                .trim();
+    }
+
+    /**
+     * Truncates a String to the given length.
+     * With ellipses.
+     *
+     * @param length The amount to add.
+     * @param string The string to add them to.
+     * @return The truncated String.
+     */
+    public static String truncate(int length, String string) {
+        return truncate(length, string, true);
+    }
+
+    /**
+     * Truncates a String to the given length.
```

Comment:
And?

Reply:
This still needs more info. Remove the line under and expand this one. Explain you can add ellipses, also recommends 3 more in length where possible if they do add ellipses.